apps/src/code-studio/components/header/EditableProjectName.jsx

```diff
@@ -64,18 +72,27 @@ class UnconnectedEditProjectName extends React.Component {
       return;
     }
 
-    dashboard.project.rename(newName, () => {
-      dashboard.header.updateTimestamp();
-      this.props.refreshProjectName();
-      this.setState({
-        savingName: false
+    dashboard.project
+      .rename(newName)
+      .then(() => {
+        dashboard.header.updateTimestamp();
+        this.props.refreshProjectName();
+        this.setState({
+          savingName: false
+        });
+        this.props.finishEdit();
+        this.setState({
+          savingName: true
+        });
+      })
+      .catch(error => {
+        if (error instanceof NameFailureError) {
+          this.props.setNameFailure(error.nameFailure);
+        }
+        this.setState({
+          savingName: false
+        });
```

Comment:
You could put `this.setState({savingName: false})` into a [`.finally` clause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) in this promise chain, since it happens on both paths.